crates/pop-cli/src/main.rs

```diff
@@ -40,39 +45,115 @@ enum Commands {
 			#[cfg(feature = "parachain")]
-			commands::new::NewCommands::Parachain(cmd) => cmd.execute().await?,
+			new::NewCommands::Parachain(cmd) => match cmd.execute().await {
+				Ok(template) => {
+					// telemetry should never cause a panic or early exit
+					tel_data = (
+						"new",
+						"parachain",
+						json!({template.provider().unwrap_or("provider-missing"): template.name()}),
+					);
+					Ok(())
+				},
+				Err(e) => Err(e),
+			},
 			#[cfg(feature = "parachain")]
-			commands::new::NewCommands::Pallet(cmd) => cmd.execute().await?,
+			new::NewCommands::Pallet(cmd) => {
+				// when there are more pallet selections, this will likely have to move deeper into the stack
+				tel_data = ("new", "pallet", json!("template"));
+
+				cmd.execute().await
+			},
 			#[cfg(feature = "contract")]
-			commands::new::NewCommands::Contract(cmd) => cmd.execute().await?,
-		}),
+			new::NewCommands::Contract(cmd) => {
+				// When more contract selections are added this will likely need to go deeper in the stack
+				tel_data = ("new", "contract", json!("default"));
+
+				cmd.execute().await
+			},
+		},
 		Commands::Build(args) => match &args.command {
 			#[cfg(feature = "parachain")]
-			commands::build::BuildCommands::Parachain(cmd) => cmd.execute(),
+			build::BuildCommands::Parachain(cmd) => {
+				tel_data = ("build", "parachain", json!(""));
+
+				cmd.execute()
+			},
 			#[cfg(feature = "contract")]
-			commands::build::BuildCommands::Contract(cmd) => cmd.execute(),
+			build::BuildCommands::Contract(cmd) => {
+				tel_data = ("build", "contract", json!(""));
+
+				cmd.execute()
+			},
 		},
 		#[cfg(feature = "contract")]
-		Commands::Call(args) => Ok(match &args.command {
-			commands::call::CallCommands::Contract(cmd) => cmd.execute().await?,
-		}),
-		Commands::Up(args) => Ok(match &args.command {
+		Commands::Call(args) => match &args.command {
+			call::CallCommands::Contract(cmd) => {
+				tel_data = ("call", "contract", json!(""));
+
+				cmd.execute().await
+			},
+		},
+		Commands::Up(args) => match &args.command {
 			#[cfg(feature = "parachain")]
-			commands::up::UpCommands::Parachain(cmd) => cmd.execute().await?,
+			up::UpCommands::Parachain(cmd) => {
+				tel_data = ("up", "parachain", json!(""));
+
+				cmd.execute().await
+			},
 			#[cfg(feature = "contract")]
-			commands::up::UpCommands::Contract(cmd) => cmd.execute().await?,
-		}),
+			up::UpCommands::Contract(cmd) => {
+				tel_data = ("up", "contract", json!(""));
+
+				cmd.execute().await
+			},
+		},
 		#[cfg(feature = "contract")]
 		Commands::Test(args) => match &args.command {
-			commands::test::TestCommands::Contract(cmd) => cmd.execute(),
+			test::TestCommands::Contract(cmd) => match cmd.execute() {
+				Ok(feature) => {
+					tel_data = ("test", "contract", json!(feature));
+					Ok(())
+				},
+				Err(e) => Err(e),
+			},
 		},
+	};
+
+	// Best effort to send on first try, no action if failure
+	let _ =
```

Comment:
As above, wrap these two records in `if let Some(t) = maybe_tel` to only spawn if telemetry enabled.